FRAuth/FRAuth/DeviceBinding/KeyAware.swift

```diff
@@ -0,0 +1,102 @@
+// 
+//  KeyAware.swift
+//  FRAuth
+//
+//  Copyright (c) 2022 ForgeRock. All rights reserved.
+//
+//  This software may be modified and distributed under the terms
+//  of the MIT license. See the LICENSE file for details.
+//
+
+
+import Foundation
+import CommonCrypto
+import FRCore
+
+///Helper struct to generate and sign the keys
+struct KeyAware {
+    var userId: String
+    var timeout = 60
+    private var keySize = 256
+    private var keyType = kSecAttrKeyTypeECSECPrimeRandom
+    private var key: String
+    
+    
+    /// Initializes KeyAware with given userId String
+    /// - Parameter userId: user id for which key pair is to be generated
+    init(userId: String) {
+        self.userId = userId
+        self.key = KeyAware.getKeyAlias(keyName: userId)
+    }
+    
+    
+    /// Query attributes dictionary for generating key pair
+    func keyBuilderQuery() -> [String: Any] {
+        var query = [String: Any]()
+        
+        query[String(kSecAttrKeyType)] = String(keyType)
+        query[String(kSecAttrKeySizeInBits)] = keySize
+        
+        var keyAttr = [String: Any]()
+        keyAttr[String(kSecAttrIsPermanent)] = true
+        keyAttr[String(kSecAttrApplicationTag)] = key
+        query[String(kSecPrivateKeyAttrs)] = keyAttr
+        
+#if !targetEnvironment(simulator)
```

Comment:
if the secure key is not available , should we fail or guide the users ? something to think of here